Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -140,39 +140,34 @@ public async Task PostParticipant(int participantID)
                 //No defined end date for creation
             });
             await _context.SaveChangesAsync();
+            var username = GetAccountByParticipantID(participantID).AD_Username;
+            var participant = GetParticipantByUsername(username);
+            return participant;
         }
-        public async Task UpdateParticipant(ParticipantPatchViewModel updatedParticipant)
+        public async Task<ParticipantActivityCreatedViewModel> UpdateParticipantActivity(string username, ParticipantActivityPatchViewModel updatedParticipant)
         {
-            int participantID = Int32.Parse(_accountService.GetAccountByUsername(updatedParticipant.Username).GordonID);
-            if (updatedParticipant.ActivityID is not null)
-            {
-                var participantActivity = _context.ParticipantActivity
-                                            .FirstOrDefault(pa => pa.ParticipantID == participantID 
-                                                && pa.ActivityID == updatedParticipant.ActivityID);
-                var priv = _context.PrivType
-                                .FirstOrDefault(pt => pt.Description == updatedParticipant.ActivityPrivType);
+            int participantID = Int32.Parse(_accountService.GetAccountByUsername(username).GordonID);
+           
+            var participantActivity = _context.ParticipantActivity
+                                        .FirstOrDefault(pa => pa.ParticipantID == participantID 
+                                            && pa.ActivityID == updatedParticipant.ActivityID);
+            var priv = _context.PrivType
+                            .FirstOrDefault(pt => pt.Description == updatedParticipant.ActivityPrivType);
```

Comment:
This is giving me a little bit of code smell. If an object has a user-friendly description that is different than the unique identifier, that should probably only be used in the user-facing program, not the API. Here, it seems you are using the description of the priv type to uniquely identify a priv type.

If the description is a code (i.e. 'ADMIN' or something like that), then you should think about just making that code the unique identifier of the table & object rather than having a separate unique id.

Reply:
this is an issue im trying to figure out. I do not really want to be passing strange ID numbers to the front end but I feel like, for reliability purposes, we might have to. Perhaps I might need to make a lookup service type thing, or status/type lookup route? Do you have any opinions on this @EjPlatzer?

Reply:
I don't know how @EjPlatzer feels about this example, but if you look at involvement membership type codes, you see the useful string values that are still used as identifiers such as 'MEMBR' & 'ADMIN'. Perhaps you could just use letter codes and define them in the UI as an enum. Even if you didn't want to do that, an enum would be a good way to connect descriptions with a numeric identifier in the UI before sending it to the api.

Reply:
I agree with Bennett that you should only have one unique identifier max. The obvious options do seem to be a string and an int.

Personally, I don't see any advantage to using a string unique identifier. It seems like you're worried about code clarity, but in that case an Enum - as Bennett mentioned - seems like the obvious solution. In general, using strings as unique keys is unfortunate because of performance effects (although those are likely to be minor in this case), but also because of the fragility of string comparison - all it takes is one user accidentally typing the string wrong and viola: major bug!

Reply:
I realized after talking to Cam about workflow that the only real need of this would be when we have a drop down on the "update" resources that we would need to put in either the name/ID so I'm going to make a few routes for the for getting all types/statuses and this will solve our issue.